Sources/NIO/SelectorEpoll.swift

```diff
@@ -171,12 +171,11 @@ extension Selector: _SelectorBackendProtocol {
             ready = Int(try Epoll.epoll_wait(epfd: self.selectorFD, events: events, maxevents: Int32(eventsCapacity), timeout: -1))
         }
 
-        // start with no deregistrations happened
-        self.deregistrationsHappened = false
-        // temporary workaround to stop us delivering outdated events; possibly set in `deregister`
-        for i in 0..<ready where !self.deregistrationsHappened {
+        for i in 0..<ready {
             let ev = events[i]
-            switch ev.data.fd {
+            let fd = Int32(UInt64(ev.data.u64 & 0x00000000FFFFFFFF))
+            let eventSequenceIdentifier = UInt32(UInt64(ev.data.u64 >> 32))
```

Comment:
We don't need the intermediary `UInt64` constructor here:

```suggestion
            let eventSequenceIdentifier = UInt32(ev.data.u64 >> 32)
```